pyseir/testing/test_strategy_simulator.py

```diff
@@ -0,0 +1,207 @@
+        the coming month.
+        Probability = minimum(1, ratio of prevalent cases in company/minimum number of cases to trigger outbreak per
+        day).
+        """
+        reduced_case_num = reduced_case_num or 0
+        to_datetime = lambda x: datetime.strptime(x, DATE_FORMAT)
+        within_next_month = [s for s in self.projection.index if to_datetime(s) >= to_datetime(self.date)
+                             and to_datetime(s) <= to_datetime(self.date) + timedelta(days=30)]
+
+        prevalent_cases = ((self.projection['I'] / self.projection['N']) * self.N).loc[
+                           within_next_month].max()
+
+        outbreak_prob = np.minimum(1, (prevalent_cases - reduced_case_num) / self.min_num_case_outbreak)
+
+        return outbreak_prob
+
+    def _calculate_infection_rates(self, freq):
+        to_datetime = lambda x: datetime.strptime(x, DATE_FORMAT)
+        within_next_month = [s for s in self.projection.index if to_datetime(s) >= to_datetime(self.date)
+                             and to_datetime(s) <= to_datetime(self.date) + timedelta(days=30)]
+
+        prevalence = (self.projection['I'] / self.projection['N']).loc[within_next_month].values
+        indices = [int(i * (30 // freq) - 1) for i in range(1, int(freq + 1))][:int(freq)]
+        prevalence = prevalence[indices]
+        return prevalence
+
+    def run(self):
+        """
+
+        """
+        param_grid = {'pcr_coverage': self.pcr_coverage,
+                      'antibody_coverage': self.antibody_coverage,
+                      'pcr_frequency': self.pcr_frequency}
+
+        results = pd.DataFrame(list(ParameterGrid(param_grid)))
+
+        results['delta_p_infected'] = results['pcr_frequency'] * results['pcr_coverage'] * self.pcr_sensitivity * \
+                                      (self.projection['I'] / self.projection['N']).loc[self.date]
+        results['delta_immunity'] = (self.projection['R'] / self.projection['N']).loc[self.date]\
+                                  * (1 - self.frac_contact_active) \
+                                  * results['antibody_coverage'] * self.antibody_sensitivity
+        # assuming cases are detected and quarantined in the middle of infectious period
+        # and spend half of their contacts at the site.
+        results['prevented_secondary_transmission'] = \
+            0.5 * self.N * self.Rt.loc[self.date] * results['delta_p_infected'] * self.relative_contact_rate / 2
+        results['prevented_secondary_transmission'] *= (1 + results['delta_immunity'])
+        results['delta_covid_index'] = results['prevented_secondary_transmission'] \
+                                     * self.projection['IHR__per_capita'].loc[self.date]
+        results['test_cost_pcr'] = self.N * results['pcr_coverage'] * self.pcr_cost * results['pcr_frequency']
+
+        outbreak_prob_no_testing = self._calculate_outbreak_prob(reduced_case_num=0)
```

Comment:
This ideally includes a baseline testing rate reduction? I suppose we could make the case that that is zero since those symptomatic cases would be likely to stay home.

Reply:
Currently I assume zero testing rate outside the site if that is what you refer as baseline testing rate. But I am not sure what do you mean by "baseline testing rate reduction"? What causes the reduction?